### DISCUSSION TITLE
A lot of TV channels from Uruguay in this page (but tokenized)

### DISCUSSION THREAD

**Discussion by Carlinhos027** [iptv-org/community#1191] [Q&A] [answered]
https://tv.vera.com.uy/envivo

(In the left side you can choose the channel you want)

Is there anyway to put them here or decode them?

For example. This is the m3u8 of Canal 4 Salto:

https://vxlive.cdn.antel.net.uy/auth_V_Canal_4_Salto,vxttoken=cGF0aFVSST0lMkZhdXRoX1ZfQ2FuYWxfNF9TYWx0byUyRnZ4Zm10JTNEaGxzJTJGJTJBJmV4cGlyeT0xNjY5NzY2ODEyJnJhbmRvbT1ObnpRM1RudUZlJmMtaXA9MTc2Ljg2LjEyMC4xNzcmaWRfcmVmZXJlbmNpYT0maWRfZGlzcG9zaXRpdm89MCZpZF9mcm9udGVuZD0xMTk2JnB1YmxpY19pZD0yczZwNCw5MmIyZDlkOWY5NDIyNDBlZTdjZmFhMzc3OGJmODBmMGQxZmRmNDNlOWNkYTEzZjUxOTVjY2Y4MWMyNjExODU2/vxfmt=hls/playlist.m3u8

**Comment by BellezaEmporium** [comment, marked as answer]
in case someone wants to work on it : 

- POST with https://veratv-be.vera.com.uy/api/sesiones data "tipo=anonima", send request and keep 'token' and 'jwt'.

TOKEN will be kept over the GET query parameters, JWT will be used as a Bearer Authorization. (you'll need it as a header for further requests)

- GET all channels by doing https://cds-frontend.vera.com.uy/api-contenidos/listas/68?token=*token*, you'll get an amazing list with the IDs.

- Choose your stream and GET https://veratv-be.vera.com.uy/api/setup?token=*token*&public_id=*channelID* , search on "suggested" for the streams, link is there.
